packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/definition/ecc-trs-filer.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  attr,
+  observable,
+  customElement,
+  FASTElement,
+} from "@microsoft/fast-element";
+import { template } from "./ecc-trs-filer.template.js";
+import { styles } from "./ecc-trs-filer.styles.js";
+
+@customElement({
+  name: "ecc-client-elixir-trs-filer",
+  template,
+  styles,
+})
+export class TRSFiler extends FASTElement {
+  @attr public baseUrl = "";
+  @attr public isOpenModal = false;
+
+  @observable toolClasses: {
+    description: string;
+    id: string;
+    name: string;
+  }[] = [];
+
+  @observable createToolForm = {
+    aliases: [""],
+    checker_url: "",
+    description: "",
+    has_checker: true,
+    name: "",
+    organization: "",
+    toolclass: {
+      description: "string",
+      id: "string",
+      name: "string",
+    },
+    versions: [],
+  };
+
+  /**
+   * Load tool classes from the server
+   * @returns {Promise<void>}
+   * @async
+   */
+  async loadTools() {
```

Comment:
you're querying the server for the tools three times whenever the page is refreshed, once here, once in the "trs.list.ts" file and again in the "trs-classes.ts" file, which is inefficient, especially if the data being queried starts to get large. Is there a particular reason you're doing this?